### PR TITLE
Changed "Manage Autolab" button from a page redirect to a dropdown

### PR DESCRIPTION
**Pull request by KesterTan**
## Description
- Deleted show page which included 5 separate links.
- Changed "Manage Autolab" from a redirect button into a dropdown list

### Screenshot of the new dropdown list on desktop
Screenshot 2023-02-01 at 11 17 49 PM

### Screenshot of the new dropdown list on mobile
Screenshot 2023-02-02 at 11 38 26 AM

## Motivation and Context
Originally, clicking the button redirects the user to this page with five links. 

Screenshot 2023-02-01 at 11 33 18 PM

There's no value in having those links in a separate page since they are only five links. Hence, the dropdown.

## How Has This Been Tested?
Enter developer login below the login page.
Screenshot 2023-02-02 at 11 46 29 AM

Click the Manage Autolab button on the top right hand corner
Screenshot 2023-02-02 at 11 46 56 AM

Change the browser width to view the change in the mobile sidebar as well.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] I have run rubocop for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting
- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://github.com/autolab/docs)
- [ ] I have updated the documentation accordingly, included in this PR

### CODE REVIEW

**Comment by 20wildmanj** [comment]
Can you update your PR so that it removes any RSpec tests for `show`?

**Comment by damianhxy** [comment]
You should fully remove any code related to the admin#show method.

In `config/routes.rb`:
Replace L70 with: `resource :admin, :except => [:show] do`

In `controllers/admins_controller.rb`:
Remove L11-12:
```ruby
  action_auth_level :show, :administrator
  def show; end
```

File: app/views/layouts/_navbar.html.erb [hunk 1]
```diff
@@ -60,17 +93,14 @@
           <li><%= link_to "<i class='material-icons left'>clear_all</i> Jobs".html_safe, [@course, :jobs], title: "View a history of autograding jobs" %></li>
 
           <% if @cud.instructor? %>
-            <li><%= link_to "<i class='material-icons left'>settings</i> Manage Course".html_safe, [:manage, @course], title: "Perform course-wise administrative actions" %></li>
+            <li><%= link_to "<i class='material-icons left'>settings</i> Course".html_safe, [:manage, @course], title: "Perform course-wise administrative actions" %></li>
```
Comment (damianhxy) [inline app/views/layouts/_navbar.html.erb:96]: Why this change?

File: app/views/layouts/_navbar.html.erb [hunk 2]
```diff
@@ -40,6 +40,39 @@
       <%= link_to "Log out", destroy_user_session_path, :method => :delete  %>
     </li>
   </ul>
+
+  <% if current_user && current_user.administrator? %>
```
Comment (damianhxy) [inline app/views/layouts/_navbar.html.erb:44]: Nit: Can be simplified to `current_user&.administrator?`

File: app/views/layouts/_navbar.html.erb [hunk 3]
```diff
@@ -60,17 +93,14 @@
...
+            <li><%= link_to "<i class='material-icons left'>settings</i> Course".html_safe, [:manage, @course], title: "Perform course-wise administrative actions" %></li>
           <% end %>
         <% end %>
 
-        <% if current_user && current_user.administrator? %>
-          <li><%= link_to "<i class='material-icons left'>lock</i> Manage Autolab".html_safe, admin_path, title: "Perform system-wide administrative actions" %></li>
-        <% end %>
-
         <% if user_signed_in? %>
+          <li><a class="dropdown-trigger" href="#!" data-target="dropdown2" data-beloworigin="true" data-hover="false"><i class='material-icons left'>lock</i>Manage Autolab<i class="material-icons right">arrow_drop_down</i></a></li>
```
Comment (damianhxy) [inline app/views/layouts/_navbar.html.erb:101]: You are displaying this button even for non-admins. You should check that they are an admin first.

**Comment by damianhxy** [comment]
Currently, the "Manage Autolab" text flashes white every time you click on it.

The offending piece of code seems to be `navbar.css` L12-15 (aka remove this code):

```css
nav a, nav a:hover, nav a:focus{
  text-decoration: none;
  color:white;
}
```

Which is unnecessary because the items already have another style applied
Screenshot 2023-02-04 at 22 29 40

**Comment by damianhxy** [comment]
Right now, if a user is signed out, the sidebar nav shows no options at all:
Screenshot 2023-02-04 at 22 28 23

The problem is that the highlighted piece of code is within the `<% if user_signed_in? %> ... <% end %>` block when it shouldn't be
Screenshot 2023-02-04 at 22 32 19

RECOMMENDATION: Move the highlighted chunk of code outside of the block.

**Comment by damianhxy** [comment]
Currently, clicking on the dropdown simply triggers it open again, rather than closes it. Is there a way to change this behavior?

**Review by damianhxy — CHANGES_REQUESTED** [review]
Left some comments

File: app/views/layouts/_navbar.html.erb (hunk 1 above)
Comment (KesterTan) [reply app/views/layouts/_navbar.html.erb:96]: Changed it back.

File: db/schema.rb [hunk 4]
```diff
@@ -96,8 +96,8 @@
     t.text "embedded_quiz_form_data"
```
Comment (damianhxy) [inline db/schema.rb:96]: Could you remove your changes from this file?

File: Gemfile [hunk 5]
```diff
@@ -161,11 +161,11 @@ gem 'sprockets-rails', '>=3.2.1'
...
-# For getting file types 
+# For getting file types
 gem 'mimemagic', '>= 0.3.7'
 
 # For encrypting API tokens
 gem 'lockbox'
 
 # to decode / verify jwts for LTI Integration
-gem "jwt"
+gem "jwt"
```
Comment (damianhxy) [inline Gemfile:171]: Avoid stripping newlines

File: app/views/layouts/_navbar.html.erb (hunk 2 above)
Comment (damianhxy) [reply app/views/layouts/_navbar.html.erb:44]: This has not been addressed  yet

File: app/views/layouts/_navbar.html.erb [hunk 6]
```diff
@@ -121,8 +153,8 @@
           <% end %>
         <% end %>
 
-        <% if current_user && current_user.administrator? %>
-          <li><%= link_to "<i class='material-icons left'>lock</i> Manage Autolab".html_safe, admin_path, title: "Perform system-wide administrative actions" %></li>
+        <% if current_user&.administrator? %>
+          <li><a class="dropdown-trigger" href="#!" data-target="dropdown3" data-beloworigin="true" data-hover="false"><i class='material-icons left'>lock</i>Manage Autolab<i class="material-icons right">arrow_drop_down</i></a></li>
         <% end %>
```
Comment (damianhxy) [inline app/views/layouts/_navbar.html.erb:158]: Unaddressed comment from Joey:

Can you update your PR so that it removes any RSpec tests for admins/show?

Comment (damianhxy) [reply app/views/layouts/_navbar.html.erb:158]: See `/spec/controllers/admins_controller_spec.rb`

**Review by damianhxy — CHANGES_REQUESTED** [review]
Thanks for resolving some comments, but some were unaddressed and some new issues were introduced

File: db/schema.rb (hunk 4 above)
Comment (KesterTan) [reply db/schema.rb]: Fixed

File: app/views/layouts/_navbar.html.erb [hunk 7]
```diff
@@ -60,15 +93,14 @@
...
 
-        <% if current_user && current_user.administrator? %>
-          <li><%= link_to "<i class='material-icons left'>lock</i> Manage Autolab".html_safe, admin_path, title: "Perform system-wide administrative actions" %></li>
+        <% if current_user&.administrator? %>
+          <li><a class="dropdown-trigger" href="#!" data-target="dropdown2" data-beloworigin="true" data-hover="false"><i class='material-icons left'>lock</i>Manage Autolab<i class="material-icons right">arrow_drop_down</i></a></li>
         <% end %>
-
-        <% if user_signed_in? %>
+        <% if user_signed_in?%>
```
Comment (damianhxy) [inline app/views/layouts/_navbar.html.erb:103]: Nit: don't strip this space

**Review by damianhxy — APPROVED** [review]
- Left a nit about spacing
- Iffy dropdown behavior persists, but it is not a big deal and can be resolved in a future PR if necessary (also, only happens when I enable touch simulation on firefox, not sure if it actually affects mobile)

lgtm
